show categorization on asset logs response

kenneth051 opened the pull request:
[Finishes #166728438]

#### What does this PR do?
show categorization on asset logs response


#### Description of Task to be completed?
an asset log will show details about the asset it belongs to like asset category, asset make, asset type and asset subcategory type


#### How should this be manually tested?
n postman, go to http://127.0.0.1:8000/api/v1/asset-logs You will see more asset details of that specific asset log.



#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2146417/stories/166728438

kenneth051 committed:
show categorization on asset logs response

[Finishes #166728438]

kenneth051 committed:
show categorization on asset logs response

[Finishes #166728438]

kmazi commented on api/tests/test_asset_log.py:
```diff
@@ -784,3 +784,27 @@ def test_authenticated_admin_user_gets_filtered_list_of_asset_logs_by_asset_make
         self.assertEqual(asset_make, self.asset_make.name)
+
+    @patch("api.authentication.auth.verify_id_token")
+    def test_authenticated_admin_user_asset_logs_by_return_detailed_fields(
```
You can use a more descriptive name for this test method. Rather than writing out a new test, you could update an existing test for getting assetlogs to show that the categorization fields are included in the response.

kmazi replied:
Will those fields be returned also when viewing a single log(detail)? It would be nice to show that in the test also

raheemazeezabiodun replied:
cool cool

kenneth051 replied:
@kmazi yes, they will be

kenneth051 committed:
modifying changes

kenneth051 committed:
modifying changes

RonKbS committed:
#165376778 Add fields to user and asset models. (#301)

* feat(department): add fields to user and asset models

- add foreign key fields to users and assets to department table
- add table for asset owners and link it to asset table
- add active_inactive and paid_postpaid fields to asset table
- restrict users to view assets in their locations and departments only

[Finishes #165376778]

* feat(department): add fields to user and asset models

- adjust current_owner and previous_owner fields to current_assignee and
    previous_assignee

[Finishes #165376778]

* feat(department): add fields to user and asset models

- move department restriction to manage_asset_viewset
- change paid_postpaid field to prepaid_postpaid

[Finishes #165376778]

* feat(department): add fields to user and asset models

- add restriction for which types of assets can have active and
    prepaid_postpaid fields edited

[Finishes #165376778]

* - add expiry_date field to asset_serializer

[Finishes #165376778]

* - reformat migration file to follow recently added migrations

* - change prepaid_postpaid field to paid field with associated options
- change atmcard to embusecard

kenneth051 committed:
Merge branch 'ft-show-categorization-asset-logs-response-166728438' of https://github.com/AndelaOSP/art-backend into ft-show-categorization-asset-logs-response-166728438

raheemazeezabiodun commented on api/serializers/assets.py:
```diff
@@ -130,6 +140,39 @@ def to_internal_value(self, data):
+
+        asset_type = instance.model_number.asset_make.asset_type.name
+        paid = validated_data.get('paid')
+        active_inactive = validated_data.get('active')
```
this seems confusing, can you give it a more declarative name, how about `is_active`?

What's the rationale behind the value of this variable?

kenneth051 replied:
@raheemazeezabiodun That is not part of this PR. I have a commit from develop that has mixed up with my commits...

raheemazeezabiodun replied:
cool cool

raheemazeezabiodun commented on api/serializers/assets.py:
```diff
@@ -130,6 +140,39 @@ def to_internal_value(self, data):
+            },
+        }
+
+        if paid or expiry_date or (active_inactive is False or active_inactive is True):
```
`active_inactive is False or active_inactive is True` this will always return True as long as `active_inactive` is a boolean value.

There is no need for `active_inactive is False or active_inactive is True`

Just do this to catch when active_inactive is None
```
if paid or expiry_date or active_inactive
```

kenneth051 replied:
@raheemazeezabiodun That is not part of this PR. I have a commit from develop that has mixed up with my commits...

raheemazeezabiodun replied:
Alright

Although that part needs to be fixed, can you include it to your pr?

kenneth051 replied:
@raheemazeezabiodun No
